libs/platform/ebpf_epoch.c

```diff
@@ -452,6 +453,12 @@ ebpf_epoch_allocate_work_item(_In_ void* callback_context, _In_ const void (*cal
     work_item->callback_context = callback_context;
     work_item->header.entry_type = EBPF_EPOCH_ALLOCATION_WORK_ITEM;
 
+    if (ebpf_allocate_preemptible_work_item(
+            &work_item->preemptible_work_item, work_item->callback, work_item->callback_context) != EBPF_SUCCESS) {
+        ebpf_free(work_item);
+        return NULL;
+    }
+
     return work_item;
 }
 
```

Comment:
@Alan-Jowett - is it the responsibility of the caller that allocated the `ebpf_epoch_work_item` to call `ebpf_epoch_free_work_item` from inside the `callback` ?